pkg/kubelet/kubelet.go

```diff
@@ -1990,6 +1990,15 @@ func (kl *Kubelet) SyncPod(ctx context.Context, updateType kubetypes.SyncPodType
 	result := kl.containerRuntime.SyncPod(sctx, pod, podStatus, pullSecrets, kl.backOff)
 	kl.reasonCache.Update(pod.UID, result)
 	if err := result.Error(); err != nil {
+		// If we have any image pull errors, report any failed pull secrets
+		for _, r := range result.SyncResults {
+			if r.Error == images.ErrImagePull {
+				if len(failedPullSecrets) > 0 {
+					kl.recorder.Eventf(pod, v1.EventTypeWarning, "FailedToRetrieveImagePullSecret", "Unable to retrieve some image pull secrets: %v", failedPullSecrets)
+				}
+			}
+		}
```

Comment:
```suggestion
		// If we have any image pull errors, report any failed pull secrets
		if len(failedPullSecrets) > 0 {
			for _, r := range result.SyncResults {
				if r.Error == images.ErrImagePull {
					kl.recorder.Eventf(pod, v1.EventTypeWarning, "FailedToRetrieveImagePullSecret", "Unable to retrieve some image pull secrets: %v", failedPullSecrets)
				}
			}
		}
```

Reorder the code to make the loop happen only if the `failedPullSecrets` is not empty.

Reply:
Good shout, thanks for the review.